src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -21,7 +38,23 @@
 
     public abstract ExecutionResult execute(ExecutionContext executionContext, GraphQLObjectType parentType, Object source, Map<String, List<Field>> fields);
 
-    protected ExecutionResult resolveField(ExecutionContext executionContext, GraphQLObjectType parentType, Object source, List<Field> fields) {
+    protected ExecutionResult resolveField(final ExecutionContext executionContext, final GraphQLObjectType parentType, final Object source, final List<Field> fields) {
+        final ExecutionConstraints executionConstraints = executionContext.getExecutionConstraints();
+        return executionConstraints.callTrackingDepth(new Callable<ExecutionResult>() {
+            @Override
+            public ExecutionResult call() throws Exception {
+                return resolveFieldImpl(executionConstraints, executionContext, parentType, source, fields);
+            }
+        });
+    }
+
+    private ExecutionResult resolveFieldImpl(ExecutionConstraints executionConstraints, ExecutionContext executionContext, GraphQLObjectType parentType, Object source, List<Field> fields) {
+        //
+        // we return null if the max depth has been exceeded to protect against large or malicious queries
+        if (executionConstraints.hasExceededMaxDepthConstraint()) {
```

Comment:
Yeah the library does not have a good way for a consumer to "inject" a strategy in for handling errors.